src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedValueAssignmentTests.cs

```diff
@@ -318,6 +318,67 @@ void M()
             }.RunAsync();
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedValues)]
+        public async Task Assignment_ConstantValue_NoReads_TopLevel_DiscardOption()
+        {
+            var source = @"
+int x;
+{|IDE0059:x|} = 1;
+
+System.Console.WriteLine();
+";
+            // The code fix offered doesn't make much sense. The variable 'x' is still there.
+            var fixedSource = @"
+int x;
+_ = 1;
+
+System.Console.WriteLine();
+";
+
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources = { source },
+                    OutputKind = OutputKind.ConsoleApplication,
+                },
+                FixedCode = fixedSource,
+                Options =
+                {
+                    { CSharpCodeStyleOptions.UnusedValueAssignment, UnusedValuePreference.DiscardVariable },
+                },
+                LanguageVersion = LanguageVersion.CSharp9,
+
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedValues)]
+        public async Task Assignment_ConstantValue_NoReads_TopLevel_UnusedLocalVariableOption()
+        {
+            var source = @"
+int x;
+x = 1; // BUG! No diagnostic is produced
```

Comment:
Let me know if this should be fixed in this PR, or we can fix it later as a separate issue.

Reply:
i'm fine with changing it in another PR/issue.